velox/exec/NestedLoopJoinProbe.cpp

```diff
@@ -435,40 +428,120 @@ RowVectorPtr NestedLoopJoinProbe::getNextCrossProductBatch(
     const std::vector<IdentityProjection>& probeProjections,
     const std::vector<IdentityProjection>& buildProjections) {
   VELOX_CHECK_GT(buildVector->size(), 0);
-  std::vector<VectorPtr> projectedChildren(outputType->size());
-  size_t numOutputRows = 0;
 
   // If it's a cross join and there is a single build record, we use the entire
   // probe batch `input_` and the single build record wrapped as a constant.
-  if (isCrossJoin() && buildRowCount_ == 1) {
-    numOutputRows = input_->size();
+  if (isCrossJoin() && isSingleBuildRow()) {
```

Comment:
What if we remove `isCrossJoin` checks here?  Should have only pure positive improvement, no?

Reply:
Good point. This is only used to generate the filter input (for non-cross-joins), not the actual output, so it should be safe.